### PR TITLE
Reader Profile Page

### PR DESCRIPTION
**Pull request by rishithab48**
- Added old profile page items to the newer one.
- Added and fixed the book stacks arrangement in the profile and book stacks page.
- Added a symbol button for removing books from book list instead of text button. (On profile page and book list page)
- Added a new logged out page.

**Fixed**
- Old profile page route.
- Removed home button on logged in version of homePage.
- Fixed some book list issues to show other users.

### CODE REVIEW

**Comment by hello202320** [comment]
I think we should of an profile page component --> author page or reader page to minimize code. Like OOP inheritance.

**Comment by rishithab48** [comment]
So, make profile class the base class and author, reader profiles as their child classes?

File: src/App.js [hunk 1]
```diff
@@ -97,7 +97,8 @@ const App = () => {
           {/* <Route path="/addbook" element={<AddBook/>}/> */}
           <Route path="/SearchResultsBook" element={<SearchResultsBook />} />
           <Route path="/ReaderQandA/:aid" element={<ReaderQandA />} />
-
+          <Route path='/loggedout' element = {<LoggedOutPage />}/>
```
Comment (racecn) [inline src/App.js:100]: Are these the permanent URLs we'd like to have? Or should we set the rendering choice within the home page component like before?

File: src/components/BookStacks/DisplayBookLists.js [hunk 2]
```diff
@@ -182,21 +195,41 @@ export const DisplayBookListItems = ({ bookData, userAuth }) => {
...
+                      <FontAwesomeIcon
+                        icon={faCircleMinus}
+                        style={{ color: "#e41111" }}
+                      />
+                    </button>
+                  </div>
                   {book ? (
                     <div key={index}>
                       <img
+                        style={{ width: "96px", height: "144px" }}
```
Comment (racecn) [inline src/components/BookStacks/DisplayBookLists.js]: Have to be careful with pixel specific styling, maybe like 10em by 15 em?

File: src/components/BookStacks/DisplayBookLists.js [hunk 3]
```diff
@@ -209,40 +242,222 @@ export const DisplayBookListItems = ({ bookData, userAuth }) => {
...
-                          )
+                  <div className="flex flex-col ml-5 mt-3"></div>
+                </div>
+              ))}
+              {books.length > 3 && (
+              <div className="col-span-5">
+              <div className="mt-4 text-center">
+                <button
+                  onClick={toggleShowAll}
+                  className="bg-maroon hover:bg-opacity-50 text-white font-bold py-2 px-4 rounded"
```
Comment (racecn) [inline src/components/BookStacks/DisplayBookLists.js]: can we make a color in the tailwind config that represents this 50% opacity maroon color, I think we are using it in a few other places like the UploadPFP component

File: src/pages/Settings.js [hunk 4]
```diff
@@ -261,48 +261,35 @@ export default function Settings() {
...
                                     </button>
+
                                     {usernameChangeRequested && (
                                         <p className="text-green-500 mt-2">Username change requested!</p>
                                     )}
                                 </div> {/* End of username change section */}
     
                                 <button
-                                    className="bg-red-500 hover:bg-red-600 text-white font-bold py-2 px-4 rounded mt-4"
+                                    className="bg-maroon hover:bg-opacity-70 text-white font-bold py-2 px-4 rounded-full mt-4"
```
Comment (racecn) [inline src/pages/Settings.js]: Can we make a tailwind entry for this color, like light-maroon?

File: src/functions/Profile/ProfilePage.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            this.pronouns = docSnapshot.data().pronouns;
+            this.username = docSnapshot.data().username;
+            console.log("username from class :",this.username);
+            this.followers = docSnapshot.data().followers;
+            this.following = docSnapshot.data().following;
+            this.followersCount = this.userData.followers?.length || 0;
+            this.followingCount = this.userData.following?.length || 0;
+            this.isFollowing = this.userData.following?.includes(this.uid);
+            //    userData.followers?.includes(currentUser.uid)
+            console.log("user data fetched successfully");
```
Comment (racecn) [inline src/functions/Profile/ProfilePage.js:123]: These logs seem like they are performing the same action as the try catch function that you have wrapping this get func

File: src/functions/AddBook.js [hunk 6]
```diff
@@ -8,12 +8,14 @@ export const AddBookToFirestore = async (BookData,navigate) => {
   if (BookData){
   const isbn = GetISBN13(BookData);
   const book_volumeInfo = BookData.volumeInfo;
+  const book_saleInfo = BookData.saleInfo;
```
Comment (racecn) [inline src/functions/AddBook.js:11]: Is sale info the price? or is this the location where you can buy it?

File: src/components/user/SelectGenres.js [hunk 7]
```diff
@@ -79,7 +79,7 @@ function SelectGenres() {
 
   return (
     <>
-      <div className="flex flex-col rounded-lg shadow-2xl items-center px-4 pt-7 pb-16 mt-6 max-w-full text-xl text-center text-black shadow-sm bg-maroon bg-opacity-50 w-[908px] max-md:px-5">
+      <div className="flex flex-col rounded-lg shadow-2xl items-center px-4 pt-7 pb-16 mt-6 max-w-full w-[908px] text-xl text-center text-black shadow-sm bg-maroon bg-opacity-50 max-md:px-5">
```
Comment (racecn) [inline src/components/user/SelectGenres.js]: can we make this a non-pixel based width?

File: src/functions/AddBook.js (hunk 6 above)
Comment (rishithab48) [reply src/functions/AddBook.js:11]: saleInfo has the price information and purchase links from google books. Mutahhara is working on it.

File: src/pages/Settings.js (hunk 4 above)
Comment (rishithab48) [reply src/pages/Settings.js]: I added the color to tailwing.config.js and changed it here.

File: src/components/user/SelectGenres.js (hunk 7 above)
Comment (rishithab48) [reply src/components/user/SelectGenres.js]: I changed it to rem

File: src/components/BookStacks/DisplayBookLists.js (hunk 3 above)
Comment (rishithab48) [reply src/components/BookStacks/DisplayBookLists.js]: I found the color and added it to tailwind.config.js. UploadPFP has a different shade of it. I will find the color and change it.

File: src/components/BookStacks/DisplayBookLists.js (hunk 2 above)
Comment (rishithab48) [reply src/components/BookStacks/DisplayBookLists.js]: Changed it to rem.

File: src/App.js (hunk 1 above)
Comment (rishithab48) [reply src/App.js:100]: No, I am just using it to have a separate page when a user logged out and redirect them to logged out home page.

File: src/functions/Profile/ProfilePage.js (hunk 5 above)
Comment (rishithab48) [reply src/functions/Profile/ProfilePage.js:123]: I removed the else. Try-catch is used to catch any exceptions in this case.

File: src/App.js (hunk 1 above)
Comment (rishithab48) [reply src/App.js:100]: logged out page is to make sure that users logged out properly and to get back to the logged out navigation bar.

**Pull request merged**